backend/src/yeomanui.ts

```diff
@@ -157,8 +159,42 @@ export class YeomanUI {
     }
   }
 
+  private updateGeneratorStarted() {
+    if (!_.isNil(this.startTime)) {
+      this.logger.trace("Start time was already initialized", {startTime: this.startTime, generatorName: this.generatorName});
+      return;
+    }
+
+    this.startTime = Date.now();
+    const eventType = EVENT_TYPES.PROJECT_GENERATION_STARTED;
+    let customEvents = [this.generatorName];
+    getSWA().track(eventType, customEvents);
+    this.logger.trace("SAP Web Analytics tracker was called and start time was initialized", {
+      eventType, generatorName: this.generatorName, startTime: this.startTime, customEvents});
+  }
+
+  private updateGeneratorEnded(eventType: string, errorMessage?: string) {
+    if (_.isNil(this.startTime)) {
+      this.logger.error("Start generation time was not initialized");
+      return;
+    }
+
+    const endTime = Date.now();
+    const generationTimeMilliSec = endTime - this.startTime;
+    const generationTimeSec = Math.round(generationTimeMilliSec/1000);
+    let customEvents = [this.generatorName, generationTimeSec.toString()];
+    if (!_.isNil(errorMessage)) {
+      customEvents.push(errorMessage);
+    }
+    getSWA().track(eventType, customEvents);
+    this.logger.trace("SAP Web Analytics tracker was called", 
+      {eventType, generatorName: this.generatorName, generationTimeSec, generationTimeMilliSec, endTime, startTime: this.startTime, customEvents, errorMessage});
+  }
+
```

Comment:
I recommend putting this code at /swa-tracker/swa-tracker-wrapper.ts  or at /swa-tracker/swa-tracker-utils.ts

Reply:
Interesting idea.